docs/user-guide.md

```diff
@@ -15,37 +15,34 @@ Such annotations are useful to (for example):
 
 ## Java variables are references
 
-In Java, all variables, except primitives like `int`, are references. We often
-think of `String x;` as meaning "`x` is a `String`", but actually it means "`x` is a
-_reference_ to a `String`". One important difference between these two views is that
-`x` can also be null. The goal of JSpecify is to move towards a world where `String
-x;` means "`x` is a reference to an actual string", and if you want a reference
-that might be null then you write `@Nullable String x;`.
+In Java, all non-primitive variables are references. We often think of `String
+x;` as meaning "`x` is a `String`", but actually it means "`x` is a _reference_
```

Comment:
OK. I did actually take note of your suggestion to use "string" rather than "object of type `String`" everywhere else but I left this one in. But I agree that we can be more precise.